RadixWallet/Clients/OnLedgerEntitiesClient/OnLedgerEntitiesClient+Interface.swift

```diff
@@ -391,6 +391,32 @@ extension OnLedgerEntitiesClient {
 }
 
 extension OnLedgerEntitiesClient {
+	/// Returns the validator of a correctly linked LSU, and `nil` for any other resource
+	public func isLiquidStakeUnit(_ resource: OnLedgerEntity.Resource) async -> OnLedgerEntity.Validator? {
+		guard let validatorAddress = resource.metadata.validator?.asGeneral else {
+			return nil
+		}
+
+		// Fetch validator info
+		let validator = try? await getEntity(
+			validatorAddress,
+			metadataKeys: .liquidStakeUnitMetadataKeys,
+			cachingStrategy: .useCache,
```

Comment:
Hmm I think this should not `useCache` but the `neverUseCache` or what I called it, since it uses the metadata keys set `liquidStakeUnitMetadataKeys` which differs from the "standard one". So for this resource we have already fetched metadata with the standard key set, so the metadata values we need will not be in the cache.

And by using `useCache` we are overwriting the cache with non standard metadata values, thus purging it, from data we might need to populate Asset details for it?

I might be mistaken, but just wanted to check if you have thought about this and tested this?

Reply:
I presumed the cache key would include the keys, but I can look at it

Reply:
Just be sure to use the same metadata keys as when fetching the validator details in Account details.

Reply:
I dont think the cache key contain metadatakeys no, I think the cache key is the validatorAddress?